### ISSUE TITLE
Parallel make (-j12) failure

### ISSUE THREAD

**Issue by jherland** [systemd/systemd#1577]
On my Arch Linux box with GNU Make 3.82, building via the official instructions:

```
./autogen.sh
./configure CFLAGS='-g -O0 -ftrapv' --enable-compat-libs --enable-kdbus --sysconfdir=/etc --localstatedir=/var --libdir=/usr/lib
make -j12
[...]
  CC       src/journal/libsystemd_journal_internal_la-journal-authenticate.lo
  CC       src/journal/libsystemd_journal_internal_la-fsprg.lo
  CCLD     libudev-internal.la
  CC       libsystemd_journal_la-libsystemd-journal.lo
libtool: link: `src/libudev/libudev.lo' is not a valid libtool object
make[2]: *** [libudev-internal.la] Error 1
make[2]: *** Waiting for unfinished jobs....
  CC       libsystemd_login_la-libsystemd-login.lo
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```

But building with `make -j1` works:

```
make -j1
[...]
  CC       src/libudev/libudev.lo
[...]
  CCLD     libudev-internal.la
[...]
```

It seems there is a missing dependency from `libudev-internal.la` to `src/libudev/libudev.lo` that (in my case) causes former target to be executed before the latter target is ready.

Has anybody else seen this? Is building with -jN (for N > 1) supported by systemd?

**Comment by ronnychevalier** [comment]
Yes, building with -jN is supported.

I tried couple of times and I was not able to reproduce this issue with current master (bf3ee9c), a clean build, and -j12.

**Comment by dvdhrm** [comment]
This doesn't make any sense to me, either. libudev-internal is just a static archive, I don't see how it can lack dependencies. Anyway, can you tell use the systemd version you're using? Also, does this still happen with -git?

**Comment by jherland** [comment]
This happens when trying to build current master branch of systemd. I first discovered it while trying to build v224. That said, I'm pretty sure it is caused by some misconfiguration or similar on this specific machine, as I am able to build it successfully (with -j12) on my laptop (also Arch Linux, but with GNU make v4.1). Seeing that, I just now upgraded GNU make (from v3.82) to v4.1 on my troubled machine, and that seems to have resolved the problem. So that points towards a potential bug in GNU make. Still weird, though, as systemd continues to build fine on our other machines (CentOS w/GNU make v3.82)...
